src/beast/beast/nudb/detail/format.h

```diff
@@ -188,6 +217,34 @@ data_size (std::size_t size, std::size_t key_size)
         size;                   // Data
 }
 
+// Returns the closest power of 2 not less than x
+template <class = void>
+std::size_t
+ceil_pow2 (unsigned long long x)
+{
+    static const unsigned long long t[6] = {
+        0xFFFFFFFF00000000ull,
+        0x00000000FFFF0000ull,
+        0x000000000000FF00ull,
+        0x00000000000000F0ull,
+        0x000000000000000Cull,
+        0x0000000000000002ull
+    };
+
+    int y = (((x & (x - 1)) == 0) ? 0 : 1);
+    int j = 32;
+    int i;
+
+    for(i = 0; i < 6; i++) {
+        int k = (((x & t[i]) == 0) ? 0 : j);
+        y += k;
+        x >>= k;
+        j >>= 1;
+    }
+
+    return std::size_t(1)<<y;
+}
+
```

Comment:
Compulsory addition of my own favorite version of this function and my typical rant that clz (count leading zeroes) is not already in the std::lib:
````
return x > 1 ? std::numeric_limits<std::size_t>::digits - clz(x-1) : 0;
````